### PR TITLE
nft: L-13 and M-08

### PR DESCRIPTION
**Pull request by sam-goldman**
L-13: Refactors a lot of the functionality in the NFT bridges into a single base contract.

M-08: Fixes upgradeability inconsistencies in the ERC721 bridge contracts.

The third commit replaces `CrossDomainEnabled` with `CrossDomainMessenger` in the ERC721 bridge contracts because `CrossDomainEnabled` is not upgradeable.

The fourth commit makes `otherBridge` and `messenger` in the NFT bridge contracts immutable as requested [here](https://github.com/ethereum-optimism/optimism/pull/3459).

### CODE REVIEW

**Comment by sam-goldman** [comment]
Added a commit that re-orders the `OptimismMintableERC721Created` event as mentioned [here](https://github.com/ethereum-optimism/optimism/pull/3104#discussion_r973046221)

File: packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    CrossDomainMessenger
+} from "@eth-optimism/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title ERC721Bridge
+ * @notice ERC721Bridge is a base contract for the L1 and L2 ERC721 bridges.
+ */
+abstract contract ERC721Bridge is Initializable {
```
Comment (tynes) [inline packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol:14]: Does this need to be `Initializable` now? I see `initialize()` was removed

Comment (tynes) [reply packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol:14]: I think its safe to remove `Initializable`

Comment (sam-goldman) [reply packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol]: Fixed

**Review by tynes — APPROVED** [review]

File: packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol (hunk 1 above)
Comment (maurelian) [reply packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol]: Looks like it's still there?

Comment (sam-goldman) [reply packages/contracts-periphery/contracts/universal/op-erc721/ERC721Bridge.sol]: I think you may have been looking at an old commit in this PR. It's been removed in the [latest commit](https://github.com/ethereum-optimism/optimism/pull/3491/files#diff-6cb5db984e0919f9f53eca29b6996b78c8213f875cef4bd516b0a765fcd01a58)

**Review by tynes — CHANGES_REQUESTED** [review]
This needs a new rebase

**Review by maurelian — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Pull request merged**